aws/request/request_retry_test.go

```diff
@@ -0,0 +1,107 @@
+package request
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+)
+
+func r(t *testing.T, url string) *http.Request {
+	r, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		t.Fatalf("can't forge request: %v", err)
+	}
+	return r
+}
+
+func Test_shouldRetryCancel_nil(t *testing.T) {
+	shouldRetryCancel(nil)
```

Comment:
Is it possible for a nil value to be provided to `shouldRetryCancel`? 

This test should also assert the default behavior of nil error. I'm guessing a nil call to shouldRetryCancel should be false return value. But we should take a look at any code path that could provide nil to shouldRetryCancel too to see if that can be prevented.